makefiles/tools/avrdude.inc.mk

```diff
@@ -6,10 +6,16 @@ DEBUGSERVER_FLAGS = "-g -j usb :$(DEBUGSERVER_PORT)"
 DEBUGGER_FLAGS = "-x $(RIOTBOARD)/$(BOARD)/dist/gdb.conf $(ELFFILE)"
 DEBUGGER = $(DIST_PATH)/debug.sh $(DEBUGSERVER_FLAGS) $(DIST_PATH) $(DEBUGSERVER_PORT)
 
-# make the flasher port configurable (e.g. with atmelice the port is usb)
-# defaults to terminal's serial port if not configured
-AVRDUDE_PORT    ?= $(PORT)
-PROGRAMMER_FLAGS = -P $(AVRDUDE_PORT) $(FFLAGS_EXTRA)
+PROGRAMMER_FLAGS = -p $(subst atmega,m,$(CPU))
+
+# Set flasher port only for programmers that require it
+ifneq (,$(filter $(PROGRAMMER),arduino buspirate stk500v1 stk500v2 wiring))
+  # make the flasher port configurable (e.g. with atmelice the port is usb)
+  # defaults to terminal's serial port if not configured
+  AVRDUDE_PORT    ?= $(PORT)
+  PROGRAMMER_FLAGS += -P $(AVRDUDE_PORT)
```

Comment:
oops wait, haven't tested that: are you sure this should be `+=`? If so please align the above `?=`

Reply:
looks like `+=` is good, so please realign the `?=` to make it look nice(r)

Reply:
Done. Shall I squash?

Reply:
yep, please!